src/Microsoft.ML.StandardLearners/FactorizationMachine/FieldAwareFactorizationMachineModelParameters.cs

```diff
@@ -42,10 +42,30 @@ private static VersionInfo GetVersionInfo()
                 verReadableCur: 0x00010001,
                 verWeCanReadBack: 0x00010001,
                 loaderSignature: LoaderSignature,
-                loaderAssemblyName: typeof(FieldAwareFactorizationMachinePredictor).Assembly.FullName);
+                loaderAssemblyName: typeof(FieldAwareFactorizationMachineModelParameters).Assembly.FullName);
         }
 
-        internal FieldAwareFactorizationMachinePredictor(IHostEnvironment env, bool norm, int fieldCount, int featureCount, int latentDim,
+        public FieldAwareFactorizationMachineModelParameters(IHostEnvironment env, bool norm, int fieldCount, int featureCount, int latentDim,
+            float[] linearWeights, float[] latentWeightsAligned) : base(env, LoaderSignature)
+        {
+            Host.Assert(fieldCount > 0);
+            Host.Assert(featureCount > 0);
+            Host.Assert(latentDim > 0);
+            Host.Assert(Utils.Size(linearWeights) == featureCount);
+            LatentDimAligned = FieldAwareFactorizationMachineUtils.GetAlignedVectorLength(latentDim);
+            Host.Assert(Utils.Size(latentWeightsAligned) == checked(featureCount * fieldCount * LatentDimAligned));
+
+            _norm = norm;
+            FieldCount = FieldCount;
+            FeatureCount = featureCount;
+            LatentDim = latentDim;
+            _linearWeights = linearWeights;
+
+            _latentWeightsAligned = new AlignedArray(featureCount * fieldCount * LatentDimAligned, 16);
+            _latentWeightsAligned.CopyFrom(latentWeightsAligned.AsSpan());
+        }
+
+        internal FieldAwareFactorizationMachineModelParameters(IHostEnvironment env, bool norm, int fieldCount, int featureCount, int latentDim,
             float[] linearWeights, AlignedArray latentWeightsAligned) : base(env, LoaderSignature)
```

Comment:
>float[] linearWeights, AlignedArray latentWeightsAligned [](start = 12, length = 56)

@wschin, are this two similar array with only difference is one of them is array and second is align array?
If yes, I don't understand why I need to pass both of them, and not just get first and construct align version.
If not, why they called this way?

Reply:
`linearWeights` is linear coefficient where SSE is not very helpful to compute related values. In contrast, `latentWeights`  is the latent vector of each feature and heavily uses SEE.

Reply:
This is all nice from internal point of view.
My question are user specific.
Does user need to provide this TWO SAME ARRAYS or not?
If not, can you formulate rule according to which we can construct align array?

---
In reply to: [242650269](https://github.com/dotnet/machinelearning/pull/1901#discussion_r242650269) [](ancestors = 242650269)

Reply:
Same array? `LinearWeights` and `latentWeights` are two independent things.